KAN-36 feat double tab teleport

RyuJeMu opened the pull request:
## What  
This PR adds enhancements and fixes related to the teleportation feature and cooldown display. It includes the following changes:  
- **Handled Key Holding:** Implemented an exception to ensure teleportation does not trigger while holding down a key.  
- **Teleport Cooldown:** Adjusted the teleportation cooldown time to 3.5 seconds.  
- **Cooldown Display:** Added a visual representation of the cooldown on the screen, including a countdown timer inside a circular indicator.  
- **Double-Tap Teleportation:** Enabled teleportation to the edges of the screen when double-tapping the respective direction keys.  
- **Distance Calculation:** Distance calculation is currently non-functional and will be addressed in a future update.  

## Why  
The teleportation feature required fine-tuning to improve usability and visual feedback. These changes ensure a smoother and more user-friendly experience, while avoiding unintended behavior such as triggering actions while holding down keys.  

## How to Test  
1. **Test Key Holding Exception:**  
   - Press and hold the right or left directional key and ensure that teleportation is not triggered.  

2. **Test Cooldown Display:**  
   - Perform a double-tap teleportation.  
   - Verify that a cooldown circle with a countdown timer appears on the screen, preventing immediate reuse of the teleport feature.  

3. **Test Cooldown Timing:**  
   - Wait for 3.5 seconds after teleportation and ensure that the feature becomes available again.  

4. **Test Edge Teleportation:**  
   - Double-tap the left key and confirm the player teleports to the left edge of the screen.  
   - Double-tap the right key and confirm the player teleports to the right edge of the screen.  

5. **Test Distance Calculation:**  
   - Attempt to calculate the distance moved by the player. Note that this feature is currently non-functional.

RyuJeMu committed:
Update sprite type based on destruction state

RyuJeMu committed:
Teleport to screen edge on right or left double-tap

RyuJeMu committed:
Added exception handling for edge cases in Double Tap functionality, ensuring robust boundary checks and preventing invalid inputs.

RyuJeMu committed:
Logic not working

RyuJeMu committed:
Cooldown and Double-Tap functionality implementation

RyuJeMu committed:
Modify cooldown-circle, add timer which shows cooldown

SEOKMINJUN commented on src/screen/GameScreen.java:
```diff
@@ -242,6 +242,17 @@ protected boolean update() {
 		super.update();
 
 		if (gameProgress) {
+			if (Globals.getInputManager().isDoubleTab(KeyEvent.VK_RIGHT) && doubleTapCooldown.getRemainingTime() == 0 ) {
```
This is not handle ship's key setting.
And ship's movement is out of there class. This is should be in ship class

SEOKMINJUN commented on src/screen/GameScreen.java:
```diff
@@ -242,6 +242,17 @@ protected boolean update() {
+				Globals.getLogger().info("Detected Double Tab Right");
+				doubleTapCooldown.reset();
+			}
+			if (Globals.getInputManager().isDoubleTab(KeyEvent.VK_LEFT) && doubleTapCooldown.getRemainingTime() == 0) {
```
same here too

RyuJeMu committed:
Add detectedDoubleTab method in Ship class, delete and some changes in GameScreen

RyuJeMu commented on src/entity/Ship.java:
```diff
@@ -401,24 +397,43 @@ public void subtractHealth(){
+			if (isRight) { this.positionX = 0; }
+	}
+
+	public void detectedDoubleTap(){
+		if (Globals.getInputManager().isDoubleTab(KeyEvent.VK_RIGHT) && doubleTapCooldown.getRemainingTime() == 0 ) {
+			moveToEdgeLeft(true);
+			Globals.getLogger().info("Detected Double Tab Right");
+			doubleTapCooldown.reset();
+		}
+		if (Globals.getInputManager().isDoubleTab(KeyEvent.VK_LEFT) && doubleTapCooldown.getRemainingTime() == 0) {
+			moveToEdgeRight(true);
+			Globals.getLogger().info("Detected Double Tab Left");
+			doubleTapCooldown.reset();
+		}
+	}
```
I add detectedDoublTap() in Ship.java. Teleport-function was executed successfully.

RyuJeMu committed:
And result of Distance-Calculation shows on game screen.

RyuJeMu committed:
Fixing a bug where double tap works while other keys are pressed

SEOKMINJUN committed:
Update double tap detect logic.
- Make double tap detecting logic independent with InputManager
- Remove double tap stuff at GameScreen, InputManager
